sdk/src/integrationTest/java/ContractFunctionParametersIntegrationTest.java

```diff
@@ -0,0 +1,2774 @@
+            .setFunction("returnBoolean", new ContractFunctionParameters().addBool(testBoolean))
+            .setQueryPayment(new Hbar(10)).execute(testEnv.client);
+
+        assertThat(response.getBool(0)).isEqualTo(testBoolean);
+    }
+
+    @Test
+    @DisplayName("Can receive bytes value from contract call")
+    void canCallContractFunctionBytesType() throws Exception {
+        var testBytes = "Test".getBytes();
+
+        var response = new ContractCallQuery().setContractId(contractId).setGas(1500000)
+            .setFunction("returnBytes", new ContractFunctionParameters().addBytes(testBytes))
+            .setQueryPayment(new Hbar(10)).execute(testEnv.client);
+
+        assertThat(response.getBytes(0)).isEqualTo(testBytes);
+    }
+
+    @Test
+    @DisplayName("Can receive bytes array value from contract call")
+    void canCallContractFunctionBytesArrayType() throws Exception {
+        byte[][] testBytes = new byte[][]{"Test1".getBytes(), "Test2".getBytes()};
+
+        var response = new ContractCallQuery().setContractId(contractId).setGas(1500000)
+            .setFunction("returnBytesArr", new ContractFunctionParameters().addBytesArray(testBytes))
+            .setQueryPayment(new Hbar(10)).execute(testEnv.client);
+
+        var responseResult = (byte[][]) response.getResult("(bytes[])").get(0);
+
+        assertThat(responseResult).isEqualTo(testBytes);
+    }
+
+    @Test
+    @DisplayName("Can receive bytes32 value from contract call")
+    void canCallContractFunctionBytes32Type() throws Exception {
+        byte[] testBytes = "Test".getBytes();
+        byte[] testBytesLen32 = new byte[32];
+        System.arraycopy(testBytes, 0, testBytesLen32, 0, testBytes.length);
+
+        var response = new ContractCallQuery().setContractId(contractId).setGas(1500000)
+            .setFunction("returnBytes32", new ContractFunctionParameters().addBytes32(testBytesLen32))
+            .setQueryPayment(new Hbar(10)).execute(testEnv.client);
+
+        assertThat(response.getBytes32(0)).isEqualTo(testBytesLen32);
+    }
+
+    @Test
+    @DisplayName("Can receive bytes32 array value from contract call")
+    void canCallContractFunctionBytes32ArrayType() throws Exception {
+        byte[] testBytes = "Test".getBytes();
+        byte[] testBytes2 = "Test2".getBytes();
+        byte[][] testBytesLen32 = new byte[2][32];
+        System.arraycopy(testBytes, 0, testBytesLen32[0], 0, testBytes.length);
+        System.arraycopy(testBytes2, 0, testBytesLen32[1], 0, testBytes2.length);
+
+        var response = new ContractCallQuery().setContractId(contractId).setGas(1500000)
+            .setFunction("returnBytes32Arr", new ContractFunctionParameters().addBytes32Array(testBytesLen32))
+            .setQueryPayment(new Hbar(10)).execute(testEnv.client);
+
+        var responseResult = (byte[][]) response.getResult("(bytes32[])").get(0);
+
+        assertThat(responseResult).isEqualTo(testBytesLen32);
+    }
+}
```

Comment:
I would add strings and boolean

Reply:
(for `getResult` function)

Reply:
@bamzedev added for strings, but noticed that Java SDK is lacking `addBoolArray` function, so test for boolean array wasn't added.

cc @rokn